layers/stateless/sl_tensor.cpp

```diff
@@ -0,0 +1,185 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "stateless/stateless_validation.h"
+#include "generated/enum_flag_bits.h"
+#include <cstdint>
+#include <unordered_set>
+#include <iostream>
+
+namespace stateless {
+
+bool Device::manual_PreCallValidateCreateTensorARM(VkDevice device, const VkTensorCreateInfoARM *pCreateInfo,
+                                                   const VkAllocationCallbacks *pAllocator, VkTensorARM *pTensor,
+                                                   const Context &context) const {
+    bool skip = false;
+
+    if (!pCreateInfo) {
+        return skip;
+    }
+    const Location create_info_loc = context.error_obj.location.dot(Field::pCreateInfo);
+    // Validation for parameters excluded from the generated validation code due to a 'noautovalidity' tag in vk.xml
+    if (pCreateInfo->sharingMode == VK_SHARING_MODE_CONCURRENT) {
+        // If sharingMode is VK_SHARING_MODE_CONCURRENT, queueFamilyIndexCount must be greater than 1
+        if (pCreateInfo->queueFamilyIndexCount <= 1) {
+            skip |= LogError("VUID-VkTensorCreateInfoARM-sharingMode-09723", device, create_info_loc.dot(Field::sharingMode),
+                             "VK_SHARING_MODE_CONCURRENT, but queueFamilyIndexCount is %" PRIu32 ".",
+                             pCreateInfo->queueFamilyIndexCount);
+        }
+
+        // If sharingMode is VK_SHARING_MODE_CONCURRENT, pQueueFamilyIndices must be a pointer to an array of
+        // queueFamilyIndexCount uint32_t values
+        if (pCreateInfo->pQueueFamilyIndices == nullptr) {
+            skip |= LogError("VUID-VkTensorCreateInfoARM-sharingMode-09722", device, create_info_loc.dot(Field::sharingMode),
+                             "is VK_SHARING_MODE_CONCURRENT, but pQueueFamilyIndices is NULL.");
+        }
+    }
+    const auto *opaque_capture = vku::FindStructInPNextChain<VkOpaqueCaptureDescriptorDataCreateInfoEXT>(pCreateInfo->pNext);
+    if (opaque_capture) {
+        // If the pNext chain includes a VkOpaqueCaptureDescriptorDataCreateInfoEXT structure, flags must contain
+        // VK_TENSOR_CREATE_DESCRIPTOR_BUFFER_CAPTURE_REPLAY_BIT_ARM
+        if (!(pCreateInfo->flags & VK_TENSOR_CREATE_DESCRIPTOR_BUFFER_CAPTURE_REPLAY_BIT_ARM)) {
+            skip |= LogError("VUID-VkTensorCreateInfoARM-pNext-09727", device, create_info_loc.dot(Field::pNext),
+                             "If the pNext chain includes a VkOpaqueCaptureDescriptorDataCreateInfoEXT structure, flags must "
```

Comment:
If you run the tests with `--print-vu` you will see why this looks strange currently`

```suggestion
                             "includes a VkOpaqueCaptureDescriptorDataCreateInfoEXT structure, but flags is missing VK_TENSOR_CREATE_DESCRIPTOR_BUFFER_CAPTURE_REPLAY_BIT_ARM"
```

Reply:
ah, yes, we found out about --print-vu a bit late, and we surely have missed some not-so-helpful error messages...